### PR TITLE
[PLGN-651] - Abnormal Security-Not parsing the timestamps provided by the user and just passing through the exact timestamp to the server

### PR DESCRIPTION
**Pull request by rbowden-r7**
### Description

Describe the proposed changes:

  - After changes made in the 2.0.0 release an issue where the wrong timeformat was getting passed to the UI, when the user entered `2023-01-01T21:11:38Z` the value of `2023-01-01T21:11:38:00:00` was being passed to the abnormal api and failing, so the plan is to just pass the users timestamp directly to the api with no parsing on our end

https://issues.corp.rapid7.com/browse/PLGN-651

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

all unit tests are still passing

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

testing using the test postmnan instance we have (note this is not fully the same as the full api, but will at least show no runtime issues)

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/abnormal_security/help.md [hunk 1]
```diff
@@ -85,7 +85,7 @@ provided, it will return up to 100 latest results
 
 |Name|Type|Default|Required|Description|Enum|Example|
 | :--- | :--- | :--- | :--- | :--- | :--- | :--- |
-|filter_key|enum - lastModifiedTime - createdTime - customerVisableTime|lastModifiedTime|False|This input enables you to select what timestamp to filter on, default is lastModifiedTime|None|lastModifiedTime|
+|filter_key|string|lastModifiedTime|False|This input enables you to select what timestamp to filter on, default is lastModifiedTime|['lastModifiedTime', 'createdTime', 'customerVisableTime', '']|lastModifiedTime|
```
Comment (dlaverty-r7) [inline plugins/abnormal_security/help.md:88]: Should this remain as an enum?

**Review by dlaverty-r7 — APPROVED** [review]

File: plugins/abnormal_security/help.md (hunk 1 above)
Comment (rbowden-r7) [reply plugins/abnormal_security/help.md:88]: This needs to be of type string as thats the type of the values in the enum comes from the enum col

File: plugins/abnormal_security/icon_abnormal_security/util/api.py [hunk 2]
```diff
@@ -91,24 +91,12 @@ def generate_filter_params(self, from_date: str = None, to_date: str = None, fil
         if from_date or to_date:
             params = {"filter": filter_key}
             if from_date:
-                params["filter"] = params["filter"] + f" gte {self.parse_date(from_date)}"
+                params["filter"] = params["filter"] + f" gte {from_date}"
```
Comment (igorski-r7) [inline plugins/abnormal_security/icon_abnormal_security/util/api.py:94]: I think that'd be better to have everything in f-strings 😄 
```suggestion
                params["filter"] = f"{params.get('filter', '')} gte {from_date}"
```

Comment (rbowden-r7) [reply plugins/abnormal_security/icon_abnormal_security/util/api.py]: This has been updated to use f-strings

**Comment by rbowden-r7** [comment]
Adding do not merge label due to code freeze

**Review by igorski-r7 — APPROVED** [review]

**Pull request merged**